### ISSUE TITLE
Missing procedure entry point "GetDpiForWindow" in DLL gw2addon_gw2radial.dll

### ISSUE THREAD

**Issue by ArchaeanDragon** [Friendly0Fire/GW2Radial#230]
Just installed v1.4.2 of GW2 Addon Manager fresh after removing all addon directories, selected only GW2Radial, hit update, then ran the game. Once through the GW2 game launcher, the missing DLL entry point error dialog pops up. Hitting OK allows the game to start, but GW2 Radial does not appear to be loaded. Shift-Alt-M doesn't do anything (but bring up the in-game map, as I would expect if the hotkeychord isn't being trapped by GW2Radial).

**To Reproduce**
1. Install GW2_UOAOM_1.4.2 clean.
2. Run it.
3. Select ONLY GW2 Radial, then click "Update"
4. It installs Addon Loader and GW2 Radial, then says it is done.
5. Close the addon manager.
6. Launch the game.
7. See the error dialog after logging in and clicking "Play". Click OK.
8. Game starts normally, but without GW2 Radial functioning.

**Expected behavior**
No error, and the addon present and working.

**Screenshots**

**System configuration (please fill in):**
 - OS: Windows 8.1 Pro
 - GPU: Radeon R9 390X
 - Chainloading, if any: Only whatever GW2 UOAOM and GW2Radial does in the latest version.

**Comment by ArchaeanDragon** [comment]
Not sure, but I think that is a Windows 10+ winuser function which isn't available in Win 8.1, so maybe it is not finding it in the loaded system DLLs, and is defaulting to the referencing DLL?

**Comment by ArchaeanDragon** [comment]
Yeah. It's recommended to use GetDpiForMonitor for Win 8.1, and GetDeviceCaps for older systems.

https://mariusbancila.ro/blog/2021/05/19/how-to-build-high-dpi-aware-native-desktop-applications/

If that helps any.

**Comment by Friendly0Fire** [comment]
While I can't release a new build right away for just this bug, I can at least provide you with a fixed version: 
[gw2addon_gw2radial.zip](https://github.com/Friendly0Fire/GW2Radial/files/7256152/gw2addon_gw2radial.zip).

This should resolve the issue and the next patch release will include it. Let me know if it helps!

**Comment by ArchaeanDragon** [comment]
It works, but had a very strange error crop up after I opened the config menu with Shift-Alt-M, and then switched to my second monitor where this web page is to type this message.

GW2Radial Fatal Error
Fatal Error: shader compilation failed! Error code was 0x88760B59. Please report..(etc)

Probably a different issue.

**Comment by ArchaeanDragon** [comment]
Now it always happens when launching the game and getting to character select screen, but before the characters are loaded.

I should also point out that I have D3DCompiler_43.dll in bin64 instead of D3DCompiler_47.dll as mentioned in your other issue on core dependencies. Curious. Not finding any good info on the version differences.

**Comment by ArchaeanDragon** [comment]
Looks like 43 is for Windows 8.1 and 47 is for Windows 10.

**Comment by ArchaeanDragon** [comment]
Was able to launch the game and get fully in-world before it popped up this time. Seems to be inconsistent when it happens.

0x88760B59 is D3DXERR_INVALIDDATA - Guess it will give more info in the trace window if you're debugging, but it might be one of "those" errors one user sees that doesn't show up in testing. Just to be on the safe side, I am going to restart the system. Never know if a driver has gone off the deep end from corrupted internal state and makes for monsters that go bump in the night.

**Comment by ArchaeanDragon** [comment]
OK. After a full restart, still getting the error after logging in and getting in-world. I removed GW2 Radial by renaming the DLL, and logged back in. The game runs fine now, so it does appear to be something GW2 Radial is doing. Going to leave it out until you have a chance to get back to me. I know you're extra busy atm with GW2 addon updates as well, so no rush. Thank you for your time! :)

**Comment by Friendly0Fire** [comment]
It looks like applications are expected to distribute the correct updated `d3dcompiler_47.dll`, so here's what we'll do. Here's the DLL: [d3dcompiler_47.zip](https://github.com/Friendly0Fire/GW2Radial/files/7256459/d3dcompiler_47.zip)

I don't know where it'll get picked up by the game, so could you please try placing it in `bin64` and, if that doesn't work, in the root directory of the game? I'll make sure the new Addon Manager installs this.

**Comment by ArchaeanDragon** [comment]
OK. So I put the _47.dll into the bin64 dir, renaming the _43 one to .tmp. Started the game. It gave the same shader error again.

I went back into the bin64 directory, and there was a new _43.dll file there. Looks like something is re-creating it if missing. I deleted the new one and copied the _47.dll to the main dir then restarted. Back in game now for a few mins, got into the Radial options, reset my hotkeys and ran through the mounts, testing. No problems. (I like the new dismount when clicking the button a second time! :D) It also still recreated a copy of the _43.dll in the bin64 dir, but it apparently is using the _47 one in the main dir, according to Process Explorer, so I think that fixed it.

I'll keep playing for a while and if I have any other issues, I'll update here this evening.

Thanks again for the speedy fixes and support! :)
